[18.0.0-rc.0] Failed to load config "plugin:@ngrx/recommended-requiring-type-checking"

### Which @ngrx/* package(s) are the source of the bug?

eslint-plugin

### Minimal reproduction of the bug/regression with instructions

I was using `ngrx` version `18.0.0-beta.1` and everything was working fine, but then I updated to the first release candidate and I got the following errors:

```sh
Failed to process project graph. Run "nx reset" to fix this. Please report the issue if you keep seeing it.
  AggregateCreateNodesError: An error occurred while processing files for the @nx/eslint/plugin plugin.
    - apps/my-project/.eslintrc.json: Failed to load config "plugin:@ngrx/recommended-requiring-type-checking" to extend from.
  Referenced from: /Users/julien.marcou/dev/my-repo/.eslintrc.angular.json
```

### Expected behavior

NgRx ESLint should work as it was in the beta 1

### Versions of NgRx, Angular, Node, affected browser(s) and operating system(s)

NgRx: 18.0.0-rc.0
Angular: 18.0.2
Nx: 19.2.2

### Other information

I'm using Nx to manage multiple projects in a monorepo, hence the specific log from Nx, but I do believe that the problem comes from NgRx

Here is my `.eslint.angular.json` file:

```json
{
  "extends": [".eslintrc.json"],
  "ignorePatterns": ["!**/*"],
  "overrides": [
    {
      "files": ["*.ts"],
      "parserOptions": {
        "createDefaultProgram": true
      },
      "extends": [
        "plugin:@nx/angular",
        "plugin:@angular-eslint/template/process-inline-templates",
        "plugin:rxjs/recommended",
        "plugin:@ngrx/recommended-requiring-type-checking"
      ],
      "rules": {
        // ...
      }
    },
    {
      "files": ["*.html"],
      "extends": ["plugin:@nx/angular-template", "plugin:prettier/recommended"],
      "rules": {
        // ...
      }
    }
  ]
}
```

Removing `"plugin:@ngrx/recommended-requiring-type-checking"` from the `extends` fixes the issue, but then I will no longer benefit from NgRx's recommended linting

### I would be willing to submit a PR to fix this issue

- [ ] Yes
- [X] No

Sorry to bother you again, but I have another problem, if I now extend from `plugin:@ngrx/all`, I have the following errors:

```sh
  1:1  error  Definition for rule '@ngrx/signal-state-no-arrays-at-root-level' was not found  @ngrx/signal-state-no-arrays-at-root-level
  1:1  error  Definition for rule '@ngrx/with-state-no-arrays-at-root-level' was not found    @ngrx/with-state-no-arrays-at-root-level
```

If I turn these rules off, it's working again, am I missing something?

--- a/projects/ngrx.io/content/guide/eslint-plugin/index.md
+++ b/projects/ngrx.io/content/guide/eslint-plugin/index.md
@@ -110,7 +110,13 @@ Instead of manually configuring the rules, there are also [preconfigured configu
 <!-- DO NOT EDIT, this table is automatically generated-->
 <!-- CONFIGURATIONS-CONFIG:START -->
 
-| Name |
-| ---- |
+| Name                                                                                                                 |
+| -------------------------------------------------------------------------------------------------------------------- |
+| [all](https://github.com/ngrx/platform/blob/main/modules/eslint-plugin/src/configs/all.json)                         |
+| [component-store](https://github.com/ngrx/platform/blob/main/modules/eslint-plugin/src/configs/component-store.json) |
+| [effects](https://github.com/ngrx/platform/blob/main/modules/eslint-plugin/src/configs/effects.json)                 |
+| [operators](https://github.com/ngrx/platform/blob/main/modules/eslint-plugin/src/configs/operators.json)             |
+| [signals](https://github.com/ngrx/platform/blob/main/modules/eslint-plugin/src/configs/signals.json)                 |
+| [store](https://github.com/ngrx/platform/blob/main/modules/eslint-plugin/src/configs/store.json)                     |
 
   <!-- CONFIGURATIONS-CONFIG:END -->

